doc/builders/special/mergeNixOSModules.section.md

```diff
@@ -0,0 +1,46 @@
+# mergeNixOSModules {#sec-merge-nixos-modules}
+
+Merge multiple NixOS modules and configuration files in to a single-file module.
+
+This is for NixOS configuration files being written to servers during creation, intended for merging configuration snippets which describe the hardware properties of the host.
```

Comment:
We need to be upfront about limitations so that newcomers don't use this if they don't have to.

```suggestion
This is for deployment automation that requires you to work with a single file. In any other situation, plain `imports = [ /*...*/ ]` is the ideal solution for merging modules. Unlike `imports`, `serializeModules` works at the file level instead of the Nix language level, so it will not preserve the meaning of path expressions. The modules must avoid Nix expressions like `./by-relative-path.conf` or rely on your automation to provide such files where the serialized modules will be copied to.

An example application is where NixOS configuration files need to be written to servers during creation, intended for merging configuration snippets which describe the hardware properties of the host.
```

It also does not support non-file module references, but I think that fact is sufficiently implied by "works at the file level".